### PR TITLE
Mroytman/update docker

### PR DESCRIPTION
**Pull request by MichaelRoytman**

### CODE REVIEW

File: README-Hacking.md [hunk 1]
```diff
@@ -10,11 +10,11 @@ Execute this command from the root of the repository:
...
+    $ docker run -p 8080:8080 -v jenkins:/edx/var/jenkins tools-jenkins
 
-In both instances ports 8080 and 50000 will be exposed to the host system, and a local volume will be shared on the 
-container. This local volume contains all of the configuration for Jenkins, and will be preserved between container
-stops and starts.
+In both instances port 8080 will be exposed to the host system, and a volume will be created on the 
+container. This volume contains all of the configuration for Jenkins, and will be preserved between container
+stops and starts, except in the case of plugin updates or installs (see WIP: Updating the Docker Image below).
 
 **NOTE:** You are not required to use `.docker-volumes/jenkins`, and can use practically any directory on your local
```
Comment (MichaelRoytman) [inline README-Hacking.md]: Should this section be remove, as we no longer mount host directories?

**Comment by MichaelRoytman** [comment]
Does Credentials section of README-Hacking.md need to be updated?

**Comment by MichaelRoytman** [comment]
When trying to build the new docker-in-docker/Dockerfile, I get the following error:

"Step 8/9 : RUN sudo apt-get install -y docker-engine
 ---> Running in ccee947b9acd
Reading package lists...
Building dependency tree...
Reading state information...
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 docker-engine : Depends: init-system-helpers (>= 1.18~) but 1.14 is to be installed
                 Depends: libdevmapper1.02.1 (>= 2:1.02.90) but 2:1.02.77-6ubuntu2 is to be installed
                 Depends: libsystemd0 but it is not installable
                 Recommends: aufs-tools but it is not going to be installed
                 Recommends: cgroupfs-mount but it is not installable or
                             cgroup-lite but it is not going to be installed
E: Unable to correct problems, you have held broken packages.
The command '/bin/sh -c sudo apt-get install -y docker-engine' returned a non-zero code: 100"

Worth looking into or fixing?

File: README-Hacking.md (hunk 1 above)
Comment (jibsheet) [inline README-Hacking.md]: Yes, remove it

File: README-Hacking.md [hunk 2]
```diff
@@ -72,10 +72,18 @@ button to expose the final two fields.
...
-the Docker container, they must be listed in ```docker/plugins.txt```. To be installed in a deployment environment,
-they must be added to the appropriate Ansible role defaults.
+The [edxops/tools-jenkins](https://hub.docker.com/r/edxops/tools-jenkins/) image is used in the Docker steps above. The required plugins have been pre-installed on the container. Feel free to install additional plugins. If you'd like to add or modify Jenkins plugins, follow the steps below.
+
+Note: Adding or modifying Jenkins plugin(s) will wipe your Jenkins workspace.
+
+Update the plugin list by changing the plugin version number(s) or by adding additional plugin(s) in the tools_jenkins Ansible role in the configuration repository on your local machine. The role is available [here](https://github.com/edx/configuration/blob/master/playbooks/roles/tools_jenkins/defaults/main.yml). Build a new Docker image that incorporates your changes by running the following command from the root of the configuration repository:
+
+	$ docker build -f docker/build/tools-jenkins/Dockerfile -t edxops/tools-jenkins:latest .
```
Comment (jibsheet) [inline README-Hacking.md]: If make docker.build.tools_jenkins tags, then we should document that.  If it doesn't, then I'm good with this.  We'll clean that up later.

**Comment by jibsheet** [comment]
I think there are still unpushed local changes here (and you'll need to rebase in order to resolve a few conflicts)

**Pull request merged**